### PR TITLE
Add Ruby 3.1 to CI matrix

### PR DESCRIPTION
**Pull request by alejandroperea**
This PR adds Ruby 3.1 to the CI matrix.

Rails version has been bumped to 7.0.1, that comes with `net-*` gems, not loaded by default in Ruby 3.1

Also was important to bump `thor` to its latest version to avoid `DidYouMean` deprecation warnings.

### CODE REVIEW

**Review by deivid-rodriguez — APPROVED** [review]
❤️

File: .github/mergify.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ pull_request_rules:
       - check-success=test(ruby-3.0, rails_60)
       - check-success=test(ruby-3.0, rails_61)
       - check-success=test(ruby-3.0, rails_70)
+      - check-success=test(ruby-3.1, rails_61)
```
Comment (javierjulio) [inline .github/mergify.yml:19]:
```suggestion
      - check-success=test(ruby-3.1, rails_60)
      - check-success=test(ruby-3.1, rails_61)
```

**Review by javierjulio — APPROVED** [review]
Thanks! This is ready, just one update is needed.

File: .github/mergify.yml (hunk 1 above)
Comment (alejandroperea) [reply .github/mergify.yml:19]: Good catch! I've missed that one. Added :)

Comment (javierjulio) [reply .github/mergify.yml:19]: Awesome thanks!

**Review by javierjulio — APPROVED** [review]
Thanks again! I'll merge.

**Pull request merged**